src/update-changelog.ts

```diff
@@ -157,6 +160,54 @@ async function getCommitHashesInRange(
   return await runCommand('git', revListArgs);
 }
 
+type AddNewCommitsOptions = {
+  mostRecentTag: string | null;
+  repoUrl: string;
+  loggedPrNumbers: string[];
+  projectRootDirectory?: string;
+};
+
+/**
+ * Get the list of new change entries to add to a changelog.
+ *
+ * @param options - Options.
+ * @param options.mostRecentTag - The most recent tag.
+ * @param options.repoUrl - The GitHub repository URL for the current project.
+ * @param options.loggedPrNumbers - A list of all pull request numbers included in the relevant parsed changelog.
+ * @param options.projectRootDirectory - The root project directory, used to
+ * filter results from various git commands. This path is assumed to be either
+ * absolute, or relative to the current directory. Defaults to the root of the
+ * current git repository.
+ * @returns A list of new change entries to add to the changelog, based on commits made since the last release.
+ */
+async function getNewChangeEntries({
+  mostRecentTag,
+  repoUrl,
+  loggedPrNumbers,
+  projectRootDirectory,
+}: AddNewCommitsOptions) {
+  const commitRange =
+    mostRecentTag === null ? 'HEAD' : `${mostRecentTag}..HEAD`;
```

Comment:
This results in a linter error: 
```
Unexpected negated condition. eslint(no-negated-condition)
```

Reply:
Hmm, we did use `null` intentionally to represent lack of a most recent tag, so the use of `null` matches the type that `getMostRecentTag` returns. However, I suppose that the tag name could be an empty string in theory if for some reason `git describe` returns an empty string, and that case we would want to fall back to `HEAD`. So, we could update `getMostRecentTag` to just return `string` instead of `string | null`, except that in case there is no recent tag, it would then `return '';`. That might look strange, but we could add a comment saying why we're doing that or give that empty string a name that describe its purpose.

In any case, it seems that this line was merely copied from another function, so maybe we want to make that change in another PR?

Reply:
I like using `null` for clear distinction and visibility, but in any case I agree with revisiting this in its own ticket since this wasn't a change introduced by this PR.